src/components/AnnouncementsModal.js

```diff
@@ -0,0 +1,305 @@
+      width: 70%;
+    `}
+  >
+    <Title>{title}</Title>
+    {imageSrc && (
+      <Link href={callToAction}>
+        <Image
+          src={imageSrc}
+          alt={callToActionLabel}
+          css={`
+            width: 100%;
+            max-height: 30vh;
+            background-position: center center;
+            background-repeat: no-repeat;
+            object-fit: cover;
+
+            &:hover {
+              opacity: 0.85;
+            }
+          `}
+        />
+      </Link>
+    )}
+
+    <div
+      dangerouslySetInnerHTML={{ __html: body }}
+      css={`
+        strong {
+          font-weight: bold;
+        }
+        a {
+          color: ${getValueForAnnouncementContent('linkColor')};
+        }
+        p {
+          margin-bottom: 1rem;
+        }
+        ol {
+          display: block;
+          list-style-type: decimal;
+          margin-block-start: 1em;
+          margin-block-end: 1em;
+          margin-inline-start: 0px;
+          margin-inline-end: 0px;
+          padding-inline-start: 40px;
+        }
+        ul {
+          display: block;
+          list-style-type: disc;
+          margin-block-start: 1em;
+          margin-block-end: 1em;
+          margin-inline-start: 0px;
+          margin-inline-end: 0px;
+          padding-inline-start: 40px;
+        }
+      `}
+    />
+    <Inline as="div" justifyContent="flex-end">
+      {callToAction && (
+        <Link href={callToAction}>
+          <Button>{callToActionLabel}</Button>
+        </Link>
+      )}
+    </Inline>
+  </Stack>
+);
+
+AnnouncementContent.propTypes = {
+  title: PropTypes.string,
+  imageSrc: PropTypes.string,
+  body: PropTypes.node,
+  callToAction: PropTypes.string,
+  callToActionLabel: PropTypes.string,
+};
+
+const AnnouncementsModal = ({
+  visible,
+  announcements,
+  seenAnnouncements,
+  setSeenAnnouncements,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+  const cookieOptions = {
+    path: '/',
+    maxAge: 60 * 60 * 24 * 30,
+  };
+  const [cookies, setCookie] = useCookies(['seenAnnouncements']);
+  const [selectedAnnouncementId, setSelectedAnnouncementId] = useState('');
+  const [selectedAnnouncement, setSelectedAnnouncement] = useState(undefined);
+
+  const addToSeenAnnouncements = (id) => {
+    setSeenAnnouncements((prev) => (prev.length > 0 ? [...prev, id] : [id]));
+    setCookie(
+      'seenAnnouncements',
+      cookies.seenAnnouncements
+        ? [...cookies.seenAnnouncements, selectedAnnouncementId]
+        : [selectedAnnouncementId],
+      cookieOptions,
+    );
+  };
+
+  useEffect(() => {
+    if (cookies.seenAnnouncements) {
+      const updatedSeenAnnouncements = cookies.seenAnnouncements.filter(
+        (announcement) => !announcements.includes(announcement.uid),
+      );
+      setSeenAnnouncements(updatedSeenAnnouncements);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!visible || announcements.length === 0) return;
+    setSelectedAnnouncementId(announcements[0].uid);
```

Comment:
Nit: The advantage of an early return is not that big here. We could also write it as
```
if (visible && announcements.length > 0) {
    setSelectedAnnouncementId(announcements[0].uid);
} 
```
Early returns are mostly useful if you have more than one check.

Or if you prefer to keep using the short syntax:
```
if (visible && announcements.length > 0) setSelectedAnnouncementId(announcements[0].uid);
```

Personally less a fan of that syntax though (also in the case of the early return).

But anyway this is all very minor.